src/pages/en/core-concepts/astro-components.md

```diff
@@ -125,22 +125,59 @@ They can be used to style your components, and all style rules are automatically
 
 ### Client-Side Scripts
 
-To send JavaScript to the browser without [using a framework component](/en/core-concepts/component-hydration) (React, Svelte, Vue, Preact, Solid...) you can use a `<script>` tag in your Astro component body.
+To send JavaScript to the browser without [using a framework component](/en/core-concepts/framework-components) (React, Svelte, Vue, Preact, SolidJS, AlpineJS, Lit...) you can use a `<script>` tag in your Astro component template and send JavaScript to the browser that executes in the global scope.
 
 ```astro
 ---
-// Your component script here!
+// Example: Using Astro with script tags
 ---
+<h1>Not clicked</h1>
+<button>Click to change heading</button>
+
 <script>
-  document.querySelector('h1').style.color = 'red';
+document.querySelector("button").addEventListener("click",() => {
+    document.querySelector("h1").innerText = "clicked"
+})
 </script>
+```
+ > ⚠️ Starting in `--experimental-static-build` (v0.23.x), you must opt in to `<script>` element processing via the `hoist` attribute:
 
-<h1>Hello, world!</h1>
+```astro
+<script>
+  // Will be rendered into the HTML exactly as written!
+  // ESM imports will not be resolved relative to the file.
+</script>
+
+<script hoist>
+  // Processed! Bundled! ESM imports work, even to npm packages.
+</script>
+```
+#### Importing Scripts
```

Comment:
Since we're not in the migration guide anymore, I would refocus this section just on "Loading External Scripts" with the Section `1.` below. 

Section `2.` can probably be brought into the section above, or given its own "`<script hoist>`" section.


I think this is what I'd recommend:

```
### Client-Side Scripts
### Using Hoisted Scripts
### Loading External Scripts
```